src/components/organisms/TimingEventModal/TimingEventModal.tsx

```diff
@@ -79,27 +79,27 @@ export const TimingEventModal: React.FC<TimingEventModalProps> = ({
   const onUpdateEvent = useCallback(
     async (data: EventInput) => {
       const start = dayjs(`${data.start_date} ${data.start_time}`);
-      const formEvent: VenueEvent = {
+      const formEvent: MaybeWithId<WorldEvent> = {
         name: data.name,
         description: data.description,
-        start_utc_seconds:
+        startUtcSeconds:
           start.unix() || Math.floor(new Date().getTime() / 1000),
-        duration_minutes:
+        durationMinutes:
           data.duration_hours * 60 + (data.duration_minutes ?? 0),
         host: data.host,
         spaceId: eventSpaceId,
         // @debt this needs figuring out. We shouldn't get to here without
         // an eventSpace
         worldId: eventSpace?.worldId ?? "",
       };
+      // Add the ID conditionally - otherwise the field is set to undefined
+      // which firebase does not like.
       if (eventSpaceId) {
         if (event?.id) {
-          // @debt this is a hack. event.venueId is the venue that contains
-          // the event inside its events subcollection. It is NOT the space
-          // that the event is being experienced in.
-          await updateEvent(event.venueId, event.id, formEvent);
+          formEvent.id = event.id;
+          await updateEvent(formEvent as WorldEvent);
```

Comment:
Because the same form is used for both creating and editing an event we *might* have an ID, but, we might not. At this point in the code we know we have an ID and so we can cast it to `WorldEvent` and pass it to `updateEvent` which needs to have an ID.

Reply:
Wouldn't it be more obvious just to add the MightHaveId generic to updateEvent's argument type?

Reply:
If you call `updateEvent` with a thing that doesn't have an ID then it's going to fail at run-time. It *must* have an ID